### ISSUE TITLE
Customizing Starcoder for FIM with Only Code Content Data

### ISSUE THREAD

**Issue by tclxmeng-jia** [bigcode-project/starcoder#125]
# Custom fine-tuning starcoder with code-only dataset

I want to use my own dataset to fine-tune starcoder. My dataset only contains the `content` code portion and does not have the `input_column_name` (prompt). 

Does finetune.py support this kind of data? It also needs to support FIM.

How can I customize the fine-tuning process to work with my code-only dataset? Any pointers appreciated!

**Comment by ArmelRandy** [comment]
Hi. If you want to fine-tune starcoder with next token prediction on just one column, you'll have to modify the code that is provided here. Namely you'll probably have to modify the function `prepare_sample_text` and other stuff. You can check out [OctoPack's repository](https://github.com/bigcode-project/octopack/blob/main/finetuning/starcoder/finetune.py) for more guidance. It supports `output_column_name = None`. Now, for FIM, please refer to this [respository](https://github.com/loubnabnl/santacoder-finetuning).

**Comment by tclxmeng-jia** [comment]
> Hi. If you want to fine-tune starcoder with masked language modeling on just one column, you'll have to modify the code that is provided here. Namely you'll probably have to modify the function and other stuff. You can check out [OctoPack's repository](https://github.com/bigcode-project/octopack/blob/main/finetuning/starcoder/finetune.py) for more guidance. It supports . Now, for FIM, please refer to this [respository](https://github.com/loubnabnl/santacoder-finetuning).`prepare_sample_text``output_column_name = None`

Really appreciate you taking the time to provide guidance and ideas, it is very valuable for me!
